client/src/main/java/com/alibaba/nacos/client/naming/NacosNamingService.java

```diff
@@ -193,13 +196,28 @@ public void registerInstance(String serviceName, String groupName, Instance inst
             beatInfo.setWeight(instance.getWeight());
             beatInfo.setMetadata(instance.getMetadata());
             beatInfo.setScheduled(false);
+            beatInfo.setTime(System.currentTimeMillis());
+            long instanceInterval = getInstanceHeartBeatInterval(instance);
+            beatInfo.setPeriod(instanceInterval == 0 ? DEFAULT_HEART_BEAT_INTERVAL : instanceInterval);
 
             beatReactor.addBeatInfo(NamingUtils.getGroupedName(serviceName, groupName), beatInfo);
         }
 
         serverProxy.registerService(NamingUtils.getGroupedName(serviceName, groupName), groupName, instance);
     }
 
+    private long getInstanceHeartBeatInterval(Instance instance) {
```

Comment:
This method should be placed in class 'Instance'. So are the methods in 'ClientBeatCheckTask'.

Reply:
ok